# Análise Exploratória

In [1]:
import sys
import numpy as np
import seaborn as sns
import pandas as pd
np.set_printoptions(threshold=sys.maxsize)
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
pd.set_option('display.max_colwidth', 80)

In [2]:
checkpoint_contratos = pd.read_csv('checkpoint_contratos_202607071250.csv')
raw_arp = pd.read_csv('raw_arp_202607071230.csv')
raw_arp_adesoes_item = pd.read_csv('raw_arp_adesoes_item_202607071231.csv', low_memory=False)
raw_arp_empenhos_saldo = pd.read_csv('raw_arp_empenhos_saldo_202607071231.csv')
raw_arp_item = pd.read_csv('raw_arp_item_202607071231.csv', low_memory=False)
raw_arp_unidades_item = pd.read_csv('raw_arp_unidades_item_202607071231.csv', low_memory=False)
raw_caracteristicas_material = pd.read_csv('raw_caracteristicas_material_202607071232.csv', low_memory=False)
raw_classe_material = pd.read_csv('raw_classe_material_202607071232.csv')
raw_classe_servico = pd.read_csv('raw_classe_servico_202607071232.csv')
raw_contratacoes = pd.read_csv('raw_contratacoes_202607071232.csv', low_memory=False,
                               dtype = {'orgaoentidadecnpj':str,
                                        'orgaosubrogadocnpj': str})
raw_contratos = pd.read_csv('raw_contratos_202607071233.csv', low_memory=False)
raw_contratos_item = pd.read_csv('raw_contratos_item_202607071234.csv', low_memory=False)
raw_divisao_servico = pd.read_csv('raw_divisao_servico_202607071234.csv')
raw_grupo_material = pd.read_csv('raw_grupo_material_202607071234.csv')
raw_grupo_servico = pd.read_csv('raw_grupo_servico_202607071235.csv')
raw_item_material = pd.read_csv('raw_item_material_202607071235.csv')
raw_item_servico = pd.read_csv('raw_item_servico_202607071235.csv')
raw_itens_contratacoes = pd.read_csv('raw_itens_contratacoes_202607071235.csv', low_memory=False,
                                     dtype = {'unidadeorgaocodigounidade':str,
                                              'orgaoentidadecnpj': str})
raw_natureza_despesa_material = pd.read_csv('raw_natureza_despesa_material_202607071235.csv')
raw_natureza_despesa_servico = pd.read_csv('raw_natureza_despesa_servico_202607071235.csv')
raw_orgao = pd.read_csv('raw_orgao_202607071236.csv',dtype = {'cnpjcpforgao':str,
                                                              'cnpjcpforgaovinculado':str,
                                                              'cnpjcpforgaosuperior':str,
                                                              'cnpjcpforgaosuperior':str})
raw_pdm_material = pd.read_csv('raw_pdm_material_202607071236.csv')
raw_resultado_itens_contratacoes = pd.read_csv('raw_resultado_itens_contratacoes_202607071236.csv', low_memory=False,
                                               dtype = {'orgaoentidadecnpj':str} )
raw_secao_servico = pd.read_csv('raw_secao_servico_202607071236.csv')
raw_subclasse_servico = pd.read_csv('raw_subclasse_servico_202607071236.csv')
raw_uasg = pd.read_csv('raw_uasg_202607071236.csv') 
# cnpjcpfuasg
raw_und_medida_servico = pd.read_csv('raw_und_medida_servico_202607071236.csv')
raw_unidade_fornecimento_material = pd.read_csv('raw_unidade_fornecimento_material_202607071236.csv')

# Domínio 5: Contratos

Contratos formalizados após os processos de contratação.




---

## `raw_contratos`
Contratos assinados entre órgãos e fornecedores.

| Coluna | Descrição |
|---|---|
| `codigounidadegestora` **PK** | Código da unidade gestora (UASG executora) |
| `numerocontrato` **PK** | Número do contrato |
| `nifornecedor` **PK** | CPF/CNPJ do fornecedor contratado |
| `codigoorgao` **FK** → `raw_orgao` | Órgão contratante |
| `idcompra` **FK** → `raw_contratacoes` | Processo de contratação de origem |
| `objeto` | Objeto do contrato |
| `valorglobal` | Valor global do contrato |
| `numeroparcelas` | Número de parcelas |
| `valorparcela` | Valor de cada parcela |
| `valoracumulado` | Valor acumulado pago |
| `datavigenciainicial` | Início da vigência |
| `datavigenciafinal` | Fim da vigência |
| `nomerazaosocialfornecedor` | Nome do fornecedor |
| `contratoexcluido` | Flag de exclusão lógica |

In [3]:
raw_contratos.shape

(378913, 39)

In [4]:
raw_contratos.duplicated().sum()

np.int64(0)

In [5]:
raw_contratos.head(3)

,codigoorgao,nomeorgao,codigounidadegestora,nomeunidadegestora,codigounidadegestoraorigemcontrato,nomeunidadegestoraorigemcontrato,receitadespesa,numerocontrato,codigounidaderealizadoracompra,nomeunidaderealizadoracompra,numerocompra,codigomodalidadecompra,nomemodalidadecompra,codigotipo,nometipo,codigocategoria,nomecategoria,codigosubcategoria,nomesubcategoria,nifornecedor,nomerazaosocialfornecedor,processo,objeto,informacoescomplementares,datavigenciainicial,datavigenciafinal,valorglobal,numeroparcelas,valorparcela,valoracumulado,totaldespesasacessorias,datahorainclusao,numerocontrolepncpcontrato,numerocontrolepncpcompra,idcompra,datahoraexclusao,contratoexcluido,unidadesrequisitantes,dt_ingest
0,26278,FUNDACAO UNIVERSIDADE FEDERAL DE PELOTAS,154047,FUNDACAO UNIVERSIDADE FEDERAL DE PELOTAS,154047,FUNDACAO UNIVERSIDADE FEDERAL DE PELOTAS,D,2025NE001010,154047.0,FUNDACAO UNIVERSIDADE FEDERAL DE PELOTAS,90068/2024,05,Pregão,99,Empenho,60,Compras,NaN,NaN,14804690000146,GABRIEL LEITZKE GOTUZZO,23110.032160/2024-51,SRP 90068/2024\r\nPEDIDO 4835 CDTEC\r\nPRAZO DE ENTREGA 30 DIAS,15404705900682024 - UASG Minuta: 154047,2025-08-15T00:00:00,2025-12-31T00:00:00,252.39,1,252.39,NaN,NaN,2025-08-26T14:42:44,NaN,92242080000100-1-000329/2024,15404705900682024,NaN,False,NaN,2026-06-28T11:16:03.275483
1,36000,MINISTERIO DA SAUDE,250103,HOSPITAL FEDERAL DE IPANEMA,250103,HOSPITAL FEDERAL DE IPANEMA,D,2024NE001809,250052.0,INSTITUTO NACIONAL DO CANCER - RJ,90007/2024,05,Pregão,99,Empenho,60,Compras,NaN,NaN,04380569000180,J R G DISTRIBUIDORA DE MEDICAMENTOS HOSPITALARES LTDA,33401093737/2024-80,EMPENHO PREGÃO 90007/2024 DO INSTITUTO NACIONAL DO CANCER - UASG 250052- ...,25005205900072024 - UASG Minuta: 250103,2024-10-10T00:00:00,2024-12-31T00:00:00,6560.82,1,6604.00,NaN,NaN,2025-05-22T19:15:30,NaN,00394544000185-1-000070/2024,25005205900072024,NaN,False,NaN,2026-06-29T13:29:20.094936
2,36000,MINISTERIO DA SAUDE,250110,SUBSECRETARIA DE ASSUNTOS ADMINISTRATIVOS,250110,SUBSECRETARIA DE ASSUNTOS ADMINISTRATIVOS,D,00064/2024,250110.0,SUBSECRETARIA DE ASSUNTOS ADMINISTRATIVOS,00389/2024,07,Inexigibilidade,50,Contrato,60,Serviços,NaN,NaN,62070115000100,INSTITUTO DOS AUDITORES INTERNOS DO BRASIL,_____.______/____-__,CONTRATAÇÃO DE 12 (DOZE) VAGAS NA AÇÃO DE DESENVOLVIMENTO: 44º CONGRESSO BRA...,NaN,2024-10-11T00:00:00,2025-04-11T00:00:00,31200.00,1,31200.00,31200.0,NaN,2024-10-14T15:38:13,00394544000185-2-001922/2024,00394544000185-1-001856/2024,25011007003892024,NaN,False,25000.112810/2024-14,2026-06-29T13:29:20.094936


### `codigoorgao`
Código do órgão responsável

In [6]:
raw_contratos.codigoorgao.value_counts(0, dropna=False).head()

codigoorgao
95556    21635
95602    17940
36000    17283
52121    14067
36201    12438
Name: count, dtype: int64

### `nomeorgao`
Nome do órgão responsável

In [7]:
raw_contratos.nomeorgao.value_counts(0, dropna=False).head()

nomeorgao
ESP-SECRETARIA DA SAUDE                          21635
ESP-SECRETARIA ADMINISTRACAO PENITENCIARIA       17940
MINISTERIO DA SAUDE                              17283
COMANDO DO EXERCITO                              14067
FUNDACAO OSWALDO CRUZ                            12438
Name: count, dtype: int64

#### Top 10 órgãos na base
| nomeorgao                                  |   proportion |
|:-------------------------------------------|-------------:|
| ESP-SECRETARIA DA SAUDE                    |         5.71 |
| ESP-SECRETARIA ADMINISTRACAO PENITENCIARIA |         4.73 |
| MINISTERIO DA SAUDE                        |         4.56 |
| COMANDO DO EXERCITO                        |         3.71 |
| FUNDACAO OSWALDO CRUZ                      |         3.28 |
| UNIVERSIDADE FEDERAL DA BAHIA              |         2.21 |
| UNIVERSIDADE FEDERAL DO PARANA             |         2.15 |
| PMSP - SECRETARIA MUNICIPAL DA SAÚDE       |         1.99 |
| JUSTICA ELEITORAL                          |         1.7  |
| UNIVERSIDADE FEDERAL DE SANTA MARIA        |         1.67 |
| ESP-SECRETARIA DA SEGURANCA PUBLICA        |         1.67 |

### `codigounidadegestora`
Código da unidade gestora

In [8]:
raw_contratos.codigounidadegestora.value_counts(1, dropna=False).head()

codigounidadegestora
153079    0.021472
925003    0.019870
153164    0.016714
153103    0.016122
150247    0.014716
Name: proportion, dtype: float64

In [9]:
raw_contratos.codigounidadegestora.value_counts(1, dropna=False).tail()

codigounidadegestora
180188    0.000003
180312    0.000003
102135    0.000003
990224    0.000003
102168    0.000003
Name: proportion, dtype: float64

### `nomeunidadegestora`
Nome da unidade gestora

In [10]:
raw_contratos.nomeunidadegestora.value_counts(1, dropna=False).head()

nomeunidadegestora
UNIVERSIDADE FEDERAL DO PARANA                 0.021472
PMSP - SECRETARIA MUNICIPAL DE SAÚDE           0.019870
UNIVERSIDADE FEDERAL DE SANTA MARIA            0.016714
UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE    0.016122
COMPLEXO HOSPITALAR E DE SAUDE DA UFBA         0.014716
Name: proportion, dtype: float64

#### Top 10 Unidades Gestoras na base de Contratos
| nomeunidadegestora                            |   proportion |
|:----------------------------------------------|-------------:|
| UNIVERSIDADE FEDERAL DO PARANA                |         2.15 |
| PMSP - SECRETARIA MUNICIPAL DE SAÚDE          |         1.99 |
| UNIVERSIDADE FEDERAL DE SANTA MARIA           |         1.67 |
| UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE   |         1.61 |
| COMPLEXO HOSPITALAR E DE SAUDE DA UFBA        |         1.47 |
| UNIVERSIDADE FEDERAL DE SANTA CATARINA        |         1.33 |
| UNIVERSIDADE FEDERAL DA PARAIBA               |         1.3  |
| HOSPITAL DAS FORCAS ARMADAS                   |         0.98 |
| UNIVERSIDADE FEDERAL DO RIO GRANDE DO SUL     |         0.91 |
| INSTITUTO FEDERAL CATARINENSE - REITORIA      |         0.87 |
| INST.FED.DE EDUC.,CIENC.E TEC.DE STA.CAT/IFSC |         0.87 |

### `codigounidadegestoraorigemcontrato`
Código da unidade gestora de origem do contrato

In [11]:
raw_contratos.codigounidadegestoraorigemcontrato.value_counts(1, dropna=False).head()

codigounidadegestoraorigemcontrato
153079    0.021472
925003    0.019870
153164    0.016714
150247    0.014716
153103    0.013763
Name: proportion, dtype: float64

In [12]:
raw_contratos.codigounidadegestoraorigemcontrato.value_counts(0, dropna=False).tail()

codigounidadegestoraorigemcontrato
180188    1
180312    1
102135    1
990224    1
102168    1
Name: count, dtype: int64

### `nomeunidadegestoraorigemcontrato`
Nome da unidade gestora de origem do contrato

In [13]:
raw_contratos.nomeunidadegestoraorigemcontrato.value_counts(1, dropna=False).head()

nomeunidadegestoraorigemcontrato
UNIVERSIDADE FEDERAL DO PARANA                 0.021472
PMSP - SECRETARIA MUNICIPAL DE SAÚDE           0.019870
UNIVERSIDADE FEDERAL DE SANTA MARIA            0.016714
COMPLEXO HOSPITALAR E DE SAUDE DA UFBA         0.014716
UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE    0.013763
Name: proportion, dtype: float64

In [14]:
raw_contratos[raw_contratos['nomeunidadegestoraorigemcontrato'] != raw_contratos['nomeunidadegestora']].shape[0]/raw_contratos.shape[0]

0.04155043506029089

In [15]:
raw_contratos[raw_contratos['nomeunidadegestoraorigemcontrato'] == raw_contratos['nomeunidadegestora']].shape[0]/raw_contratos.shape[0]

0.9584495649397091

#### Quase igual à coluna `nomenidadegestora`, apenas 4% de valores diferentes
#### (PERGUNTA) Tem essa coluna pra servir de referência ao contrato que não tem nome padronizado como no sistema?

In [16]:
raw_contratos[raw_contratos['nomeunidadegestoraorigemcontrato'] != raw_contratos['nomeunidadegestora']].head()

,codigoorgao,nomeorgao,codigounidadegestora,nomeunidadegestora,codigounidadegestoraorigemcontrato,nomeunidadegestoraorigemcontrato,receitadespesa,numerocontrato,codigounidaderealizadoracompra,nomeunidaderealizadoracompra,numerocompra,codigomodalidadecompra,nomemodalidadecompra,codigotipo,nometipo,codigocategoria,nomecategoria,codigosubcategoria,nomesubcategoria,nifornecedor,nomerazaosocialfornecedor,processo,objeto,informacoescomplementares,datavigenciainicial,datavigenciafinal,valorglobal,numeroparcelas,valorparcela,valoracumulado,totaldespesasacessorias,datahorainclusao,numerocontrolepncpcontrato,numerocontrolepncpcompra,idcompra,datahoraexclusao,contratoexcluido,unidadesrequisitantes,dt_ingest
601,26411,"INST.FED.DE EDUC.,CIENC.E TEC.DO SUDESTE MG",158123,INSTITUTO FED CIENCIA TECNOL SUDESTE MG,158414,INST FED SUDESTE DE MG/CAMPUS JUIZ DE FORA,D,2024NE000092,158123.0,INSTITUTO FED CIENCIA TECNOL SUDESTE MG,90569/2023,06,Dispensa,99,Empenho,60,Serviços,NaN,NaN,11651195000156,MINAS TIRA ENTULHO ATERRO E DESATERRO LTDA,23225.000059/2024-44,CNAE 3811-4/00 - REFERE-SE A PRESTAÇÃO DE SERVIÇOS DE LOCAÇÃO DE CAÇAMBAS PA...,15812306905692023 - UASG Minuta: 158123,2024-02-16T00:00:00,2024-12-31T00:00:00,9504.00,1,9504.00,NaN,NaN,2025-02-06T11:31:58,NaN,10723648000140-1-000203/2023,15812306905692023,NaN,False,NaN,2026-06-29T13:17:28.017786
642,36201,FUNDACAO OSWALDO CRUZ,254420,FUNDACAO OSWALDO CRUZ,254452,FUNDAÇÃO OSWALDO CRUZ,D,00050/2024,254452.0,FUNDAÇÃO OSWALDO CRUZ,00004/2023,05,Pregão,50,Contrato,48,Mão de Obra,NaN,NaN,27934344000124,SERVE MAIS TERCEIRIZACAO DE MAO DE OBRA E SERVICOS EM GERAL LTDA,25027.000236/2023-73,O OBJETO DO PRESENTE INSTRUMENTO É A CONTRATAÇÃO DE SERVIÇOS TERCEIRIZADOS E...,NaN,2024-08-12T00:00:00,2025-08-12T00:00:00,465172.68,1,465172.68,422999.52,NaN,2024-08-19T11:50:18,33781055000135-2-001135/2024,33781055000135-1-001453/2023,25445205000042023,NaN,False,NaN,2026-06-29T13:28:23.115220
708,36201,FUNDACAO OSWALDO CRUZ,254452,FUNDAÇÃO OSWALDO CRUZ,254420,FUNDACAO OSWALDO CRUZ,D,00052/2024,254452.0,FUNDAÇÃO OSWALDO CRUZ,00042/2024,06,Dispensa,98,Outros,60,Serviços,NaN,NaN,02385669000174,FIOTEC - FUNDACAO PARA O DESENVOLVIMENTO CIENTIFICO E TECNOLOGICO EM SAUDE,25027.000282/2024-53,O PRESENTE CONTRATO TEM POR OBJETO A EXECUÇÃO DAS ATIVIDADES DE APOIO LOGÍST...,NaN,2024-08-20T00:00:00,2026-08-20T00:00:00,7000000.00,1,7000000.00,7000000.00,NaN,2024-08-22T10:34:25,NaN,33781055000135-1-001463/2024,25445206000422024,NaN,False,NaN,2026-06-29T13:28:23.115220
1045,26411,"INST.FED.DE EDUC.,CIENC.E TEC.DO SUDESTE MG",158123,INSTITUTO FED CIENCIA TECNOL SUDESTE MG,158412,INST FED SUDESTE DE MG/CAMPUS RIO POMBA,D,2024NE000077,158123.0,INSTITUTO FED CIENCIA TECNOL SUDESTE MG,90009/2024,05,Pregão,99,Empenho,60,Compras,NaN,NaN,07987754000126,APIARIO E PIMENTA NOGUEIRA LTDA,23222.003285/2023-26,AQUISIÇÃO DE GÊNEROS ALIMENTÍCIOS - PREGÃO 90009/2024 - RECURSOS DO PROGRAMA...,15812305900092024 - UASG Minuta: 158123,2024-04-04T00:00:00,2025-12-31T00:00:00,22709.65,1,22709.65,22709.65,NaN,2025-01-17T14:28:55,NaN,10723648000140-1-000022/2024,15812305900092024,NaN,False,NaN,2026-06-29T13:17:28.017786
1330,37202,INSTITUTO NACIONAL DO SEGURO SOCIAL,510677,SUPERINTENDENCIA REGIONAL NORDESTE,511656,GERENCIA EXECUTIVA SANTO ANTONIO DE JESUS,D,00034/2017,511656.0,GERENCIA EXECUTIVA SANTO ANTONIO DE JESUS,00001/2017,07,Inexigibilidade,50,Contrato,60,Serviços,INCPEN,Inclusão Pendente,13504675000110,EMPRESA BAIANA DE AGUAS E SANEAMENTO SA,35536.000578/2016-78,CONTRATAÇÃO DE FORNECIMENTO DO SERVIÇODE ÁGUA E ESGOTO NA GEX SAJ E UNIDADES.,DOC SEI Nº 17708347,2024-09-19T00:00:00,NaN,56933.28,60,4744.44,3700666.80,NaN,2021-01-01T22:36:53,NaN,NaN,51165607000012017,NaN,False,NaN,2026-06-29T13:28:28.785224


### `receitadespesa`
Indica se o contrato é de receita ou despesa

In [17]:
raw_contratos.receitadespesa.value_counts(1, dropna=False).head()

receitadespesa
D    0.981299
S    0.011290
R    0.007411
Name: proportion, dtype: float64

#### Tem um terceiro valor `S`, que não está na documentação e é mais representativo (1,1%) do que os casos de Receita (0,7%)

In [18]:
raw_contratos[raw_contratos['receitadespesa'] == 'S'].head()

,codigoorgao,nomeorgao,codigounidadegestora,nomeunidadegestora,codigounidadegestoraorigemcontrato,nomeunidadegestoraorigemcontrato,receitadespesa,numerocontrato,codigounidaderealizadoracompra,nomeunidaderealizadoracompra,numerocompra,codigomodalidadecompra,nomemodalidadecompra,codigotipo,nometipo,codigocategoria,nomecategoria,codigosubcategoria,nomesubcategoria,nifornecedor,nomerazaosocialfornecedor,processo,objeto,informacoescomplementares,datavigenciainicial,datavigenciafinal,valorglobal,numeroparcelas,valorparcela,valoracumulado,totaldespesasacessorias,datahorainclusao,numerocontrolepncpcontrato,numerocontrolepncpcompra,idcompra,datahoraexclusao,contratoexcluido,unidadesrequisitantes,dt_ingest
79,26282,UNIVERSIDADE FEDERAL DE VICOSA,154051,UNIVERSIDADE FEDERAL DE VICOSA,154051,UNIVERSIDADE FEDERAL DE VICOSA,S,00067/2025,NaN,NaN,00067/2025,NAOSEAPLIC,Não se Aplica,96,Acordo de Cooperação Técnica (ACT),60,Serviços,NaN,NaN,20320503000151,FUNDACAO ARTHUR BERNARDES,23114.905926/2025-78,COOPERAÇÃO TÉCNICA ENTRE A UNIVERSIDADE E A FUNDAÇÃO DE APOIO A FIM DE POSSI...,NaN,2025-05-13T00:00:00,2030-05-13T00:00:00,50000.0,1,50000.0,50000.0,NaN,2025-05-14T08:17:16,NaN,NaN,IC000672025,NaN,False,NaN,2026-06-28T11:16:02.728347
113,26280,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,154049,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,154049,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,S,00037/2025,NaN,NaN,00000/0000,NAOSEAPLIC,Não se Aplica,96,Acordo de Cooperação Técnica (ACT),999,Internacional,NaN,NaN,546666561,FACULTAD DE FILOSOFÍA Y LETRAS DE LA UNIVERSIDAD DE BUENOS AIRES,23112.025810/2024-92,PROMOVER A MOBILIDADE DE ESTUDANTES DE GRADUAÇÃO E PÓS-GRADUAÇÃO; PROMOVER E...,NaN,2025-01-06T00:00:00,2030-01-06T00:00:00,0.0,1,0.0,NaN,NaN,2025-04-10T14:06:22,NaN,NaN,IC000000000,NaN,False,Programa de Pós-Graduação em Antropologia Social (PPGAS),2026-06-28T11:16:01.568726
341,26282,UNIVERSIDADE FEDERAL DE VICOSA,154051,UNIVERSIDADE FEDERAL DE VICOSA,154051,UNIVERSIDADE FEDERAL DE VICOSA,S,00098/2025,NaN,NaN,00098/2025,NAOSEAPLIC,Não se Aplica,98,Outros,60,Serviços,NaN,NaN,60411527000130,LABORATÓRIO BIOVET LTDA,23114.906579/2025-09,FORMALIZAR A TRANSFERÊNCIA PELA UFV À RECEPTORA DO BANCO DO ORGANISMO GENETI...,NaN,2025-06-03T00:00:00,2030-06-03T00:00:00,0.0,1,0.0,NaN,NaN,2025-06-11T11:05:26,NaN,NaN,IC000982025,NaN,False,NaN,2026-06-28T11:16:02.728347
488,26280,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,154049,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,154049,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,S,00036/2025,NaN,NaN,00000/0000,NAOSEAPLIC,Não se Aplica,96,Acordo de Cooperação Técnica (ACT),999,Internacional,NaN,NaN,546666561,FACULTAD DE FILOSOFÍA Y LETRAS DE LA UNIVERSIDAD DE BUENOS AIRES,23112.035085/2024-61,PROMOVER E DESENVOLVER NOMEADAMENTE O INTERCÂMBIO DE PROFESSORES E PESQUISAD...,NaN,2025-01-06T00:00:00,2030-01-06T00:00:00,0.0,1,0.0,NaN,NaN,2025-04-10T14:25:57,NaN,NaN,IC000000000,NaN,False,Programa de Pós-Graduação em Antropologia Social (PPGAS),2026-06-28T11:16:01.568726
489,26280,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,154049,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,154049,FUNDACAO UNIVERSIDADE FEDERAL DE SAO CARLOS,S,00001/2025,NaN,NaN,NaN,NAOSEAPLIC,Não se Aplica,96,Acordo de Cooperação Técnica (ACT),999,Internacional,NaN,NaN,Q1518001A,UNIVERSIDADE DE SANTIAGO DE COMPOSTELA,23112.040499/2024-10,DESEJANDO AS PARTES PROMOVER E FOMENTAR A CULTURA CIENTÍFICA ATRAVÉS DE UMA ...,NaN,2025-01-07T00:00:00,2029-01-07T00:00:00,0.0,1,0.0,NaN,NaN,2025-01-09T08:47:27,NaN,NaN,IC,NaN,False,Programa de Pós-Graduação em Ciência Política (PPGPol),2026-06-28T11:16:01.568726


### `numerocontrato`
Número do contrato

(PERGUNTA) esse número corresponde à qual sistema? o ComprasGov ou uma fonte?

In [19]:
raw_contratos.numerocontrato.value_counts(0, dropna=False).head()

numerocontrato
00001/2024    1866
00002/2024    1595
00003/2024    1474
00001/2025    1418
00004/2024    1386
Name: count, dtype: int64

In [20]:
raw_contratos.numerocontrato.value_counts(0, dropna=False).tail()

numerocontrato
19575/2025      1
2025NE010086    1
2025NE100857    1
2025NE100952    1
2025NE100947    1
Name: count, dtype: int64

### `codigounidaderealizadoracompra`
Código da unidade realizadora da compra

In [21]:
raw_contratos.codigounidaderealizadoracompra.value_counts(1, dropna=False).head()

codigounidaderealizadoracompra
153079.0    0.021496
925003.0    0.020005
153164.0    0.016650
153103.0    0.016241
150247.0    0.014703
Name: proportion, dtype: float64

### `nomeunidaderealizadoracompra`
Nome da unidade realizadora da compra

In [22]:
raw_contratos.nomeunidaderealizadoracompra.value_counts(1, dropna=False).head()

nomeunidaderealizadoracompra
UNIVERSIDADE FEDERAL DO PARANA                 0.021496
PMSP - SECRETARIA MUNICIPAL DE SAÚDE           0.020005
UNIVERSIDADE FEDERAL DE SANTA MARIA            0.016650
UNIVERSIDADE FEDERAL DO RIO GRANDE DO NORTE    0.016241
COMPLEXO HOSPITALAR E DE SAUDE DA UFBA         0.014703
Name: proportion, dtype: float64

#### (TO DO) Quantos % é diferente da unidade gestora e da unidade gestora da origem do contrato

### `numerocompra`
Número da compra associada ao contrato

In [23]:
raw_contratos.numerocompra.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

In [24]:
raw_contratos.numerocompra.value_counts(0, dropna=False).tail()

numerocompra
90922/2023    1
91007/2023    1
90248/2023    1
90263/2023    1
90426/2023    1
Name: count, dtype: int64

### `codigomodalidadecompra`
Código da modalidade de compra

In [25]:
raw_contratos.numerocompra.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `nomemodalidadecompra`
Nome da modalidade de compra


In [26]:
raw_contratos.nomemodalidadecompra.value_counts(1, dropna=False).head()

nomemodalidadecompra
Pregão             0.689686
Dispensa           0.171657
Inexigibilidade    0.111944
Não se Aplica      0.014959
Concorrência       0.010240
Name: proportion, dtype: float64

### `codigotipo`
Código do tipo do contrato

In [27]:
raw_contratos.codigotipo.value_counts(1, dropna=False).head()

codigotipo
99    0.723279
50    0.241148
98    0.010765
51    0.010037
96    0.005006
Name: proportion, dtype: float64

### `nometipo`
Nome do tipo do contra

In [28]:
raw_contratos.nometipo.value_counts(1, dropna=False).head()

nometipo
Empenho                               0.723279
Contrato                              0.241148
Outros                                0.010765
Credenciamento                        0.010037
Acordo de Cooperação Técnica (ACT)    0.005006
Name: proportion, dtype: float64

### `codigocategoria`
Código da categoria do contrato

In [41]:
raw_contratos.codigocategoria.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `nomecategoria`
Nome da categoria do contrato

In [41]:
raw_contratos.nomecategoria.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `codigosubcategoria`
Código da subcategoria do contrato

In [41]:
raw_contratos.codigosubcategoria.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `nomesubcategoria`
Nome da subcategoria do contrato

In [41]:
raw_contratos.nomesubcategoria.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `nifornecedor`
Identificação do fornecedor (CNPJ ou CPF)

In [41]:
raw_contratos.nifornecedor.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `nomerazaosocialfornecedor`
Nome ou razão social do fornecedor

In [41]:
raw_contratos.nomerazaosocialfornecedor.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `processo`
Número do processo administrativo do contrato

In [41]:
raw_contratos.processo.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `objeto`
Descrição do objeto do contrato

In [41]:
raw_contratos.objeto.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `informacoescomplementares`
Informações complementares do contrato

In [41]:
raw_contratos.informacoescomplementares.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `datavigenciainicial`
Data inicial da vigência do contrato

In [41]:
raw_contratos.datavigenciainicial.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `datavigenciafinal`
Data final da vigência do contrato

In [41]:
raw_contratos.datavigenciafinal.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `valorglobal`
Valor total do contrato

In [41]:
raw_contratos.valorglobal.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `numeroparcelas`
Número total de parcelas do contrato

In [41]:
raw_contratos.numeroparcelas.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `valorparcela`
Valor de cada parcela do contrato

In [41]:
raw_contratos.valorparcela.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `valoracumulado`
Valor acumulado do contrato até o momento

In [41]:
raw_contratos.valoracumulado.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `totaldespesasacessorias`
Valor total das despesas acessórias relacionadas ao contrato

In [41]:
raw_contratos.totaldespesasacessorias.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `datahorainclusao`
Data e hora de inclusão do contrato no sistema

In [41]:
raw_contratos.datahorainclusao.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `numerocontrolepncpcontrato`
Número de controle do contrato no PNCP

In [41]:
raw_contratos.numerocontrolepncpcontrato.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `numerocontrolepncpcompra`
Não está na documentação da API

In [41]:
raw_contratos.numerocontrolepncpcompra.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `idcompra`
Identificador único da compra associada ao contrato

In [41]:
raw_contratos.idcompra.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `datahoraexclusao`
Data e hora de exclusão do contrato (se aplicável)

In [41]:
raw_contratos.datahoraexclusao.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `contratoexcluido`
Indicador se o contrato foi excluído

0 – False/Não

1 – True/Sim

In [41]:
raw_contratos.contratoexcluido.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

### `unidadesrequisitantes`
Não está na documentação da API

In [41]:
raw_contratos.unidadesrequisitantes.value_counts(1, dropna=False).head()

numerocompra
90001/2024    0.020139
90002/2024    0.017038
90003/2024    0.013993
90001/2025    0.013523
90004/2024    0.012829
Name: proportion, dtype: float64

## `raw_contratos_item`
Itens detalhados de cada contrato.

| Coluna | Descrição |
|---|---|
| `codigounidadegestora` **PK/FK** → `raw_contratos` | Unidade gestora |
| `numerocontrato` **PK/FK** → `raw_contratos` | Número do contrato |
| `nifornecedor` **PK/FK** → `raw_contratos` | Fornecedor |
| `numeroitem` **PK** | Número do item no contrato |
| `contratoitemexcluido` **PK** | Flag de exclusão (faz parte da PK por design da API) |
| `idcompra` **FK** → `raw_contratacoes` | Compra de origem |
| `codigoitem` | Código no catálogo *(join com `raw_item_material`)* |
| `descricaoiitem` | Descrição do item |
| `quantidadeitem` | Quantidade contratada |
| `valorunitarioitem` | Valor unitário |
| `valortotalitem` | Valor total do item |


In [29]:
raw_contratos_item.shape

(657555, 36)

In [30]:
raw_contratos_item.duplicated().sum()

np.int64(0)

In [31]:
raw_contratos_item.head()

,codigoorgao,codigounidadegestora,codigounidadegestoraorigemcontrato,codigounidaderealizadoracompra,codigomodalidadecompra,numerocontrato,nifornecedor,nomerazaosocialfornecedor,processo,datavigenciainicial,datavigenciafinal,valorglobal,tipoitem,codigoitem,descricaoiitem,quantidadeitem,valorunitarioitem,valortotalitem,datahorainclusao,numerocontrolepncpcontrato,idcompra,datahoraexclusaocontrato,contratoexcluido,nomeorgao,nomeunidadegestora,nomeunidadegestoraorigemcontrato,nomeunidaderealizadoracompra,nomemodalidadecompra,numerocompra,datahoraexclusaoitem,contratoitemexcluido,numeroitem,esfera,poder,numerocontrolepncpcompra,dt_ingest
0,26261,153030,153030,153030.0,05,2024NE001451,30420630000186,M A 2 COMERCIAL LTDA,23088.017837/2024-56,2024-12-02T00:00:00,NaN,6171.03,Material,356558,ETANOLAMINA,1000,0.0900,90.00,2024-12-02T16:58:29,21040001000130-2-000650/2024,15303005900222024,NaN,False,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,Pregão,90022/2024,NaN,False,00027,Federal,Executivo,21040001000130-1-000139/2024,2026-06-29T13:09:28.299314
1,26261,153030,153030,153030.0,05,2024NE001451,30420630000186,M A 2 COMERCIAL LTDA,23088.017837/2024-56,2024-12-02T00:00:00,NaN,6171.03,Material,412577,HIDRÓXIDO DE POTÁSSIO,1000,0.0780,78.00,2024-12-02T16:58:29,21040001000130-2-000650/2024,15303005900222024,NaN,False,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,Pregão,90022/2024,NaN,False,00032,Federal,Executivo,21040001000130-1-000139/2024,2026-06-29T13:09:28.299314
2,26261,153030,153030,153030.0,05,2024NE001439,30420630000186,M A 2 COMERCIAL LTDA,23088.017837/2024-56,2024-12-02T00:00:00,NaN,1019.46,Material,413087,BUTÓXIDO DE TITÂNIO IV,250,3.7581,939.53,2024-12-02T16:58:30,21040001000130-2-000651/2024,15303005900222024,NaN,False,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,Pregão,90022/2024,NaN,False,00011,Federal,Executivo,21040001000130-1-000139/2024,2026-06-29T13:09:28.299314
3,26261,153030,153030,153030.0,05,2024NE001439,30420630000186,M A 2 COMERCIAL LTDA,23088.017837/2024-56,2024-12-02T00:00:00,NaN,1019.46,Material,347951,CARBONATO DE POTÁSSIO,500,0.0877,43.85,2024-12-02T16:58:30,21040001000130-2-000651/2024,15303005900222024,NaN,False,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,Pregão,90022/2024,NaN,False,00012,Federal,Executivo,21040001000130-1-000139/2024,2026-06-29T13:09:28.299314
4,26261,153030,153030,153030.0,05,2024NE001439,30420630000186,M A 2 COMERCIAL LTDA,23088.017837/2024-56,2024-12-02T00:00:00,NaN,1019.46,Material,454041,SOLUÇÃO PADRÃO,2,11.6900,23.38,2024-12-02T16:58:30,21040001000130-2-000651/2024,15303005900222024,NaN,False,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,UNIVERSIDADE FEDERAL DE ITAJUBA,Pregão,90022/2024,NaN,False,00061,Federal,Executivo,21040001000130-1-000139/2024,2026-06-29T13:09:28.299314


### `codigoorgao`
Código do órgão responsável

### `codigounidadegestora`
Código da unidade gestora do
contrato

### `codigounidadegestoraorigemcontrato`
Código da unidade gestora de origem do contrato

### `codigounidaderealizadoracompra`
Código da unidade realizadora da compra

### `codigomodalidadecompra`
Código da modalidade de compra associada ao contrato

### `numerocontrato`
Número do contrato

### `nifornecedor`
Identificação do fornecedor (CNPJ ou CPF)

### `nomerazaosocialfornecedor`
Nome ou razão social do fornecedor

### `processo`
Número do processo administrativo do contrato

### `datavigenciainicial`
Data inicial da vigência do contrato

### `datavigenciafinal`
Data final da vigência do contrato

### `valorglobal`
Valor total do contrato

### `tipoitem`
Tipo do item do contrato (ex.: material ou serviço)

### `codigoitem`
Código do item do contrato no catálogo

### `descricaoiitem`
Descrição detalhada do item do contrato

### `quantidadeitem`
Quantidade total do item contratado

### `valorunitarioitem`
Valor unitário do item contratado

### `valortotalitem`
Valor total do item no contrato

### `datahorainclusao`
Data e hora de inclusão do contrato no sistema

### `numerocontrolepncpcontrato`
Número de controle do contrato no PNCP

### `idcompra`
Identificador único da compra associada ao contrato

### `datahoraexclusaocontrato`
Data e hora de exclusão do contrato (se aplicável)

### `contratoexcluido`
Indicador se o contrato foi excluído

0 – False/Não

1 – True/Sim

### `nomeorgao`
Nome do órgão responsável

### `nomeunidadegestora`
Nome da unidade gestora do contrato

### `nomeunidadegestoraorigemcontrato`
Código da unidade gestora de origem do contrato

### `nomeunidaderealizadoracompra`
Código da unidade realizadora da compra

### `nomemodalidadecompra`
Nome da modalidade de compra associada ao contrato

### `numerocompra`
Número da compra associada ao contrato

### `datahoraexclusaoitem`
Data e hora de exclusão do item no contrato (se aplicável)

### `contratoitemexcluido`
Indicador se o item do contrato foi excluído

0 – False/Não

1 – True/Sim

### `numeroitem`
Número do item no contrato

### `esfera`

### `poder`

### `numerocontrolepncpcompra`